tests/analyzers/test_classification_performance_analyzer.py

```diff
@@ -1,41 +1,268 @@
+    assert '1' in result.reference_metrics.metrics_matrix
+    assert result.reference_metrics.confusion_matrix.labels == ['false', 'true']
+
+
+@pytest.mark.parametrize(
+    'reference_data, current_data, data_mapping',
+    (
+        # prediction and current datasets are missed
+        (
+            None,
+            None,
+            column_mapping.ColumnMapping(),
+        ),
+        # prediction dataset is missed
+        (
+            None,
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            column_mapping.ColumnMapping(),
+        ),
+        # data mapping is missed
+        (
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            None,
+        ),
+    )
+)
+def test_missed_datasets_cases(
+        analyzer: classification_performance_analyzer.ClassificationPerformanceAnalyzer,
+        reference_data: pd.DataFrame,
+        current_data: pd.DataFrame,
+        data_mapping: Optional[column_mapping.ColumnMapping],
+) -> None:
+    with pytest.raises(ValueError):
+        analyzer.calculate(
+            reference_data=reference_data,
+            current_data=current_data,
+            column_mapping=data_mapping,
+        )
+
+
+@pytest.mark.parametrize(
+    'reference_data, current_data, data_mapping',
+    (
+        # target column is missed in data mapping
+        (
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            pd.DataFrame({'target': [0, 1, 1, 0, 1]}),
+            column_mapping.ColumnMapping(target=None),
+        ),
+        # prediction column is missed in data mapping
+        (
+
+            pd.DataFrame({'target': [0, 1, 1, 0, 1]}),
+            None,
+            column_mapping.ColumnMapping(target='another_target', prediction=None),
+        ),
+        # target is incorrect in data mapping
+        (
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            column_mapping.ColumnMapping(target='another_target'),
+        ),
+        # prediction is incorrect in data mapping
+        (
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            pd.DataFrame({'target': [0, 1, 1, 0, 1], 'prediction': [1, 1, 0, 0, 1]}),
+            column_mapping.ColumnMapping(target='another_prediction'),
+        ),
+        # no data mapping but target and prediction columns have not default names
+        (
+            pd.DataFrame({'another_target': [0, 1, 1, 0, 1], 'another_prediction': [1, 1, 0, 0, 1]}),
+            pd.DataFrame({'another_target': [0, 1, 1, 0, 1], 'another_prediction': [1, 1, 0, 0, 1]}),
+            column_mapping.ColumnMapping(),
+        ),
+    )
+)
+def test_incorrect_data_mapping_in_datasets(
```

Comment:
Probably, it should be some kind of error.

Reply:
Should I change to ValueError raising?

Reply:
I think we will address this later. First, we need to figure out how we should handle such cases.